### DISCUSSION TITLE
[General] What is `fallback_oz`?

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#505] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

In the `foundry.toml` for foundry-zksync, we are encouraged to add:

```
[profile.zksync]
src = 'src'
libs = ['lib']
fallback_oz = true
mode = "3"
```

What is `mode` and what is `fallback_oz`? Where are these documented? 

### What is the specific issue or error you're encountering?

See above

### Can you share the error messages or logs you're receiving, if any?

See above

### Have you made any recent changes to the contract before encountering this issue?

See above

### Are there any external libraries or contracts that your contract interacts with?

See above

### Can you provide the relevant portions of your contract code where the issue is occurring?

See above

### Have you tried to isolate the problem, and if so, what were the results?

See above

### What steps have you already taken to try to resolve the issue?

See above

**Comment by dutterbutter** [comment, marked as answer]
This should be documented if it isn't. Will review to update the docs accordingly with a formal definition. 

This, `fallback_oz`, compiler option when enabled true will re-compile a contract in mode "z" if the size of the contract is large. Specifically, tries to recompile with `-Oz` if the bytecode is too large.

This a more optimized approach from compiling all contracts with mode "z".
